fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+
+! obtain fcst VM
+    call ESMF_GridCompGet(fcstComp, vm=fcstVM, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+! create fcst state
+    fcstState = ESMF_StateCreate(rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+! call fcst Initialize (including creating fcstgrid and fcst fieldbundle)
+    call ESMF_GridCompInitialize(fcstComp, exportState=fcstState,    &
+                                 clock=clock_fv3, userRc=urc, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    if (ESMF_LogFoundError(rcToCheck=urc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
+!
+! reconcile the fcstComp's import state
+    call ESMF_StateReconcile(fcstState, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+!
+! determine number elements in fcstState
+    call ESMF_StateGet(fcstState, itemCount=FBCount, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+    if(mype == 0) print *,'af fcstCom FBCount= ',FBcount
+!
+! allocate arrays
+    allocate(fcstFB(FBCount), fcstItemNameList(FBCount), fcstItemTypeList(FBCount))
+!
+! pull out the item names and item types from fcstState
+    call ESMF_StateGet(fcstState, itemNameList=fcstItemNameList, &
+                       itemTypeList=fcstItemTypeList, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+!
+! loop over all items in the fcstState and collect all FieldBundles
+    do i=1, FBcount
+      if (fcstItemTypeList(i) == ESMF_STATEITEM_FIELDBUNDLE) then
+        ! access the FieldBundle
+        call ESMF_StateGet(fcstState, itemName=fcstItemNameList(i), &
+                           fieldbundle=fcstFB(i), rc=rc)
+        if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+!    if(mype==0.or.mype==144) print *,'af fcstFB,i=',i,'name=',trim(fcstItemNameList(i))
+      else
+
+      !***### anything but a FieldBundle in the state is unexpected here
+          call ESMF_LogSetError(ESMF_RC_ARG_BAD,                                 &
+                                msg="Only FieldBundles supported in fcstState.", &
+                                line=__LINE__, file=__FILE__, rcToReturn=rc)
+          return
+      endif
+    enddo
+!
+!-----------------------------------------------------------------------
+!***  create and initialize Write component(s).
+!-----------------------------------------------------------------------
+!
+    if( quilting ) then
+
+!
+      allocate(wrtComp(write_groups), wrtState(write_groups) )
+      allocate(wrtFB(5,write_groups), routehandle(5,write_groups))
```

Comment:
what are these magic numbers (5) here. Shouldn't wrtFB be allocated as: `allocate(wrtFB(FBcount,write_groups))`?